src/base/QXmppJingleData.cpp

```diff
@@ -3224,3 +3224,319 @@ std::optional<QXmppJingleMessageInitiationElement::Type> QXmppJingleMessageIniti
+/// element as specified by \xep{0482, Call Invites}.
+///
+/// \ingroup Stanzas
+///
+/// \since QXmpp 1.6
+///
+
+///
+/// \brief Constructs a Call Invite element.
+///
+QXmppCallInviteElement::QXmppCallInviteElement()
+    : d(new QXmppCallInviteElementPrivate())
+{
+}
+
+///
+/// Returns the Call Invite element type.
+///
+QXmppCallInviteElement::Type QXmppCallInviteElement::type() const
+{
+    return d->type;
+}
+
+///
+/// Sets the Call Invite element type.
+///
+void QXmppCallInviteElement::setType(Type type)
+{
+    d->type = type;
+}
+
+///
+/// Returns the Call Invite element id.
+///
+QString QXmppCallInviteElement::id() const
+{
+    return d->id;
+}
+
+///
+/// Sets the Call Invite element id.
+///
+void QXmppCallInviteElement::setId(const QString &id)
+{
+    d->id = id;
+}
+
+///
+/// Returns the Call Invite element audio flag.
+///
+bool QXmppCallInviteElement::audio() const
+{
+    return d->audio;
+}
+
+///
+/// Sets the Call Invite element audio flag.
+///
+void QXmppCallInviteElement::setAudio(bool audio)
+{
+    d->audio = audio;
+}
+
+///
+/// Returns the Call Invite element video flag.
+///
+bool QXmppCallInviteElement::video() const
+{
+    return d->video;
+}
+
+///
+/// Sets the Call Invite element video flag.
+///
+void QXmppCallInviteElement::setVideo(bool video)
+{
+    d->video = video;
+}
+
+///
+/// Returns a possible Call Invite element "jingle" sub element.
+///
+std::optional<QXmppCallInviteElement::Jingle> QXmppCallInviteElement::jingle() const
+{
+    return d->jingle;
+}
+
+///
+/// Sets a possible Call Invite "jingle" sub element.
+///
+void QXmppCallInviteElement::setJingle(std::optional<Jingle> jingle)
+{
+    d->jingle = jingle;
+}
+
+///
+/// Returns possible Call Invite "external" sub elements.
+///
+std::optional<QVector<QXmppCallInviteElement::External>> QXmppCallInviteElement::external() const
+{
+    return d->external;
+}
+
+///
+/// Sets possible Call Invite "external" sub elements.
+///
+void QXmppCallInviteElement::setExternal(std::optional<QVector<External>> external)
+{
+    d->external = external;
+}
+
+/// \cond
+void QXmppCallInviteElement::parse(const QDomElement &element)
+{
+    std::optional<Type> type { stringToCallInviteElementType(element.nodeName()) };
+
+    if (!type) {
+        return;
+    }
+
+    d->type = type.value();
+    d->id = element.attribute(QStringLiteral("id"));
+
+    switch (d->type) {
+    case Type::Invite:
+        d->audio = element.attribute(QStringLiteral("audio"), QStringLiteral("true")) == QStringLiteral("true") ? true : false;
+        d->video = element.attribute(QStringLiteral("video"), QStringLiteral("false")) == QStringLiteral("true") ? true : false;
+        // fall through
+    case Type::Accept: {
+        if (const auto &jingleElement = element.firstChildElement("jingle"); !jingleElement.isNull()) {
+            if (!d->jingle) {
+                d->jingle = Jingle();
+            }
```

Comment:
I think you can just always reset the jingle element when parsing.

Reply:
Yes, what I tried to say is that you can replace `if (!d->jingle) { d->jingle = Jingle(); }` with just `d->jingle = Jingle();`. (inside the `!jingleElement.isNull()`)

Reply:
Yes and even if it is set, it's still fine to reset it here.